product_variant_cost_price/models/product.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Ainara Galdona - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import api, fields, models
+from openerp.addons import decimal_precision as dp
+
+
+class ProductProduct(models.Model):
+    _inherit = 'product.product'
+
+    @api.model
+    def _get_selection_cost_method(self):
+        return self.env['product.template'].fields_get(
+            allfields=['cost_method'])['cost_method']['selection']
+
+    def default_cost_method(self):
+        default_dict = self.env['product.template'].default_get(
+            ['cost_method'])
+        return default_dict.get('cost_method')
+
+    @api.model
+    def _set_standard_price(self, product, value):
```

Comment:
Why don't put this as a function the same that is on base, and use their inverse?